packages/tokens/src/rwa/identity_storage_registry/storage.rs

```diff
@@ -0,0 +1,502 @@
+    e.storage()
+        .persistent()
+        .get(&key)
+        .inspect(|_| {
+            e.storage().persistent().extend_ttl(
+                &key,
+                IDENTITY_TTL_THRESHOLD,
+                IDENTITY_EXTEND_AMOUNT,
+            )
+        })
+        .unwrap_or_else(|| panic_with_error!(e, IRSError::IdentityNotFound))
+}
+
+// ================ Country Profile ===================
+
+/// Retrieves a specific country profile by its index.
+///
+/// # Arguments
+///
+/// * `e` - The Soroban environment.
+/// * `account` - The account address to query.
+/// * `index` - The index of the country profile to retrieve.
+///
+/// # Errors
+///
+/// * [`IRSError::CountryProfileNotFound`] - If the index is out of bounds.
+pub fn get_country_profile(e: &Env, account: &Address, index: u32) -> CountryProfile {
+    let key = IRSStorageKey::CPEnumerable(CPEnumerableKey { account: account.clone(), index });
+    e.storage()
+        .persistent()
+        .get(&key)
+        .inspect(|_| {
+            e.storage().persistent().extend_ttl(
+                &key,
+                IDENTITY_TTL_THRESHOLD,
+                IDENTITY_EXTEND_AMOUNT,
+            )
+        })
+        .unwrap_or_else(|| panic_with_error!(e, IRSError::CountryProfileNotFound))
+}
+
+/// Retrieves the number of country profiles for a given account.
+///
+/// # Arguments
+///
+/// * `e` - The Soroban environment.
+/// * `account` - The account address to query.
+pub fn get_country_profile_count(e: &Env, account: &Address) -> u32 {
+    let key = IRSStorageKey::CPCount(account.clone());
+    e.storage()
+        .persistent()
+        .get(&key)
+        .inspect(|_| {
+            e.storage().persistent().extend_ttl(
+                &key,
+                IDENTITY_TTL_THRESHOLD,
+                IDENTITY_EXTEND_AMOUNT,
+            )
+        })
+        .unwrap_or_default()
+}
+
+/// Retrieves all country profiles for a given account.
+///
+/// # Arguments
+///
+/// * `e` - The Soroban environment.
+/// * `account` - The account address to query.
+pub fn get_country_profiles(e: &Env, account: &Address) -> Vec<CountryProfile> {
+    let count = get_country_profile_count(e, account);
+    let mut profiles: Vec<CountryProfile> = vec![e];
+
+    for index in 0..count {
+        let key = IRSStorageKey::CPEnumerable(CPEnumerableKey { account: account.clone(), index });
+        let profile: CountryProfile = e
+            .storage()
+            .persistent()
+            .get(&key)
+            .inspect(|_| {
+                e.storage().persistent().extend_ttl(
+                    &key,
+                    IDENTITY_TTL_THRESHOLD,
+                    IDENTITY_EXTEND_AMOUNT,
+                )
+            })
+            // Unwrap should be always safe, if counting is done correctly,
+            // adding "else" for consistency
+            .unwrap_or_else(|| panic_with_error!(e, IRSError::CountryProfileNotFound));
```

Comment:
I'd suggest `expect` in this case and removing the comment. That will clearly signal the intention behind the code